java-frontend/src/main/java/org/sonar/java/regex/JavaRegexSourceTrackingTextSpans.java

```diff
@@ -28,31 +28,31 @@
 import javax.annotation.Nullable;
 import org.sonar.java.AnalyzerMessage;
 import org.sonar.java.AnalyzerMessage.TextSpan;
-import org.sonar.java.regex.ast.IndexRange;
 import org.sonar.plugins.java.api.tree.LiteralTree;
+import org.sonarsource.analyzer.commons.regex.JavaRegexSource;
+import org.sonarsource.analyzer.commons.regex.ast.IndexRange;
 
-public class JavaRegexSource implements RegexSource {
-
-  private final String sourceText;
-
+public class JavaRegexSourceTrackingTextSpans extends JavaRegexSource {
```

Comment:
I'm not a huge fan of this big name. Anything simpler maybe, such as `JavaRegexTextSpans` ?

Reply:
`JavaRegexTextSpans` sounds like it's a collection of `TextSpan`s. I mean, it does contain a collection of `TextSpan`s (well, one `TextSpan` in most cases), but what it *is* is an implementation of the `RegexSource` interface for Java regular expressions that also allows you to get the `TextSpan` corresponding to each `RegexTree`. And I can't think of a shorter name that expresses that.

Reply:
`JavaAnalyzerRegexSource`